core/src/main/scala/kafka/metrics/KafkaMetricsReporter.scala

```diff
@@ -50,14 +50,16 @@ trait KafkaMetricsReporter {
 object KafkaMetricsReporter {
   val ReporterStarted: AtomicBoolean = new AtomicBoolean(false)
 
-  def startReporters (verifiableProps: VerifiableProperties) {
+  def startReporters (verifiableProps: VerifiableProperties):List[KafkaMetricsReporter] = {
+    var reporters = List[KafkaMetricsReporter]()
     ReporterStarted synchronized {
       if (!ReporterStarted.get()) {
         val metricsConfig = new KafkaMetricsConfig(verifiableProps)
         if(metricsConfig.reporters.nonEmpty) {
           metricsConfig.reporters.foreach(reporterType => {
             val reporter = CoreUtils.createObject[KafkaMetricsReporter](reporterType)
             reporter.init(verifiableProps)
+            reporters ++= List(reporter)
```

Comment:
Appending to `scala.List` is slow (prepending is fast). You should probably just use an `ArrayBuffer` and return `Seq` from the method definition.